sympy/operator/operatorop.py

```diff
@@ -0,0 +1,868 @@
+"""Provides the basic operations between the operators."""
+
+from .operator import OpExpr, Op
+from collections import defaultdict
+from sympy.core.add import Add, _addsort
+from sympy.core.mul import Mul, NC_Marker, _mulsort
+from sympy.core.parameters import global_parameters
+from sympy.core.power import Pow
+from sympy.core.singleton import S
+
+class OpAdd(OpExpr, Add):
+    """Added operator.
+
+    Examples
+    ========
+
+    >>> from sympy import sin, cos
+    >>> from sympy.abc import x
+
+    >>> 1+cos+sin
+    1 + cos + sin
+    >>> (1+cos+sin)(x)
+    sin(x) + cos(x) + 1
+    """
+    identity = S.Zero
+
+    @classmethod
+    def flatten(cls, seq):
+        from sympy.calculus.util import AccumBounds
+        from sympy.matrices.expressions import MatrixExpr
```

Comment:
It looks like this is a complete copy of `Add.flatten`. Since this class is a subclass of `Add` why is it necessary to do this?

Reply:
It's because ``Add.flatten`` sometimes returns ``Mul`` where ``OpMul`` is needed. Some parts are modified so that ``OpMul`` is returned.

Reply:
Other seemingly-copy&pasted codes are added for same reasons.

Reply:
There needs to be a better way.

Reply:
You're doing this so that you can do things like `1 + sin` but I don't think that should work anyway.

Reply:
I agree with that ``flatten`` methods in this PR needs more refinement.

Talking about ``1+sin``, I am aware of that this is ackward. However, let's think about ``OpMul`` and ``OpPow``.  
Since ``sin + cos`` is implemented, ``2*sin`` is also introduced, which will return ``2*sin(x)`` when called. In this case, ``2`` in the instance is not called - it remains intact and gets multiplied with called ``sin(x)``.  
And, since ``OpMul`` is introduced, it would be natural to introduce ``1/sin``, which is an ``OpPow`` instance and returns ``1/sin(x)`` when called. In this case, ``1`` remains intact as well.  

That is why I concluded that ``1+sin`` should be allowed, and return ``1+sin(x)``. How is your opinion?